SnapXEats/SnapXEats/Modules/RestaurantDirections/Router/RestaurantDirectionsRouter.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  RestaurantDirectionsRouter.swift
+//  SnapXEats
+//
+//  Created by synerzip on 28/02/18.
+//  Copyright © 2018 SnapXEats. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+class RestaurantDirectionsRouter {
+
+    // MARK: Properties
+
+    weak var view: UIViewController?
+    
+    private init() {}
+    static let shared = RestaurantDirectionsRouter()
+    
+    // MARK: Static methods
+
+    func loadRestaurantDirectionsModule() -> RestaurantDirectionsViewController {
+        let viewController = UIStoryboard.loadViewController() as RestaurantDirectionsViewController
+        
+        let presenter = RestaurantDirectionsPresenter()
+        let router = RestaurantDirectionsRouter()
+        let interactor = RestaurantDirectionsInteractor()
+
+        viewController.presenter =  presenter
+        
+        presenter.view = viewController
+        //presenter.baseView = viewController
```

Comment:
please check why you need to comment this line. 
are you need view in presenter .
Base View is used for some reason because of protocol oriented programming. 
if you required use both else keep baseview only it is must .